modin/core/io/column_stores/parquet_dispatcher.py

```diff
@@ -646,6 +646,110 @@ def build_index(cls, dataset, partition_ids, index_columns, filters):
                 complete_index = index_objs[0].append(index_objs[1:])
         return complete_index, range_index or (len(index_columns) == 0)
 
+    @classmethod
+    def _normalize_partitioning(cls, remote_parts, row_lengths, column_widths):
+        """
+        Normalize partitioning according to the default partitioning scheme in Modin.
+
+        The result of 'read_parquet()' is often under partitioned over rows and over partitioned
+        over columns, so this method expands the number of row splits and shrink the number of column splits.
+
+        Parameters
+        ----------
+        remote_parts : np.ndarray
+        row_lengths : list of ints or None
+            Row lengths, if 'None' won't repartition across rows.
+        column_widths : list of ints
+
+        Returns
+        -------
+        remote_parts : np.ndarray
+        row_lengths : list of ints or None
+        column_widths : list of ints
+        """
+        if len(remote_parts) == 0:
+            return remote_parts, row_lengths, column_widths
+
+        from modin.core.storage_formats.pandas.utils import get_length_list
+
+        # The code in this function is actually a duplication of what 'BaseQueryCompiler.repartition()' does,
```

Comment:
Should we explore this in a separate issue to not have the duplicate code?

Reply:
created an issue for this #7043

Reply:
@dchigarev Looks like you mean `partition_manager.rebalance_partitions`? (not `BaseQueryCompiler.repartition()`)

Reply:
no, I meant `qc.repartition()`, I tried using it and it was slower than my implementation

`partition_manager.rebalance_partitions` splits partitions one by one using `partition.mask()`, I also tried this option, and it was also slower than the current implementation

Reply:
It is, but I would rather merge it with `rebalance_partitions()` function in order to get a "unified" method to rebalance partitioning #7044